### PR TITLE
Feature/iri pipeline

### PR DESCRIPTION
**Pull request by Tinkaa**
Getting IRI into pipeline

Also tried to make tests, but they feel kinda useless as they are almost duplicate of `test_iri.py`. Let me know what you think!

### CODE REVIEW

File: src/aatoolbox/datasources/iri/iri_seasonal_forecast.py [hunk 1]
```diff
@@ -144,12 +149,21 @@ def _process(filepath: str, ds, clobber: bool):
...
-        return ds
+        try:
+            ds = xr.load_dataset(
+                self._get_raw_path(),
+                decode_times=False,
+                drop_variables="C",
+            )
+            return ds
+        except ValueError as err:
+            raise ValueError(
```
Comment (Tinkaa) [inline src/aatoolbox/datasources/iri/iri_seasonal_forecast.py:160]: @turnerm For the `iri_auth` there are two options. Either the env var is not set due to which it is None, which we catch above. However, another option is that the `iri_auth` is not valid. In that case a file is downloaded, but this actually doesn't contain any data and is not a netcdf file. I thought this was the best way to catch it, but curious to hear if you got any other ideas!

File: tests/pipeline/conftest.py [hunk 2]
```diff
@@ -7,14 +7,15 @@
...
+FAKE_IRI_AUTH = "def"
 
 
 @pytest.fixture(scope="session", autouse=True)
 def mock_aa_data_dir(session_mocker):
     """Mock out the AA_DATA_DIR environment variable."""
     session_mocker.patch.dict(
         "aatoolbox.config.pathconfig.os.environ",
-        {"AA_DATA_DIR": FAKE_AA_DATA_DIR},
+        {"AA_DATA_DIR": FAKE_AA_DATA_DIR, "IRI_AUTH": FAKE_IRI_AUTH},
```
Comment (turnerm) [inline tests/pipeline/conftest.py:18]: I know that it's annoying to make another fixture, but I feel like there should be a separate one for IRI only used in the IRI tests (that way you can control it separately if needed)

File: tests/pipeline/test_iri_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+"""Test IRI methods as called from the pipeline."""
```
Comment (turnerm) [inline tests/pipeline/test_iri_pipeline.py:1]: Can you remove some of the other IRI tests because of this file, or merge them here?

**Review by turnerm — COMMENTED** [review]
Looks good Tinka! It's making me rethink the structure of pipeline, I will mull it over during the weekend. 

The main thing is that I would maybe try to migrate some of the tests to go through the pipeline interface rather than the IRI class directly.

File: src/aatoolbox/pipeline.py [hunk 4]
```diff
@@ -101,3 +109,91 @@ def _load_codab(self, layer_name: str, clobber: bool):
...
+        GeoBoundingBox object
+
+        Examples
+        --------
+        >>> from aatoolbox.pipeline import Pipeline
+        >>> # Get boundingbox of npl codab boundaries
+        >>> pipeline = Pipeline("npl")
+        >>> npl_geobb = pipeline.load_geoboundingbox_codab()
+        """
+        gdf = self.load_codab(admin_level=0)
```
[the remark this answers is not in the document]
Comment (Tinkaa) [reply src/aatoolbox/pipeline.py:138]: Good one.. I changed the name to `load_geoboundingbox_gdf()`

File: src/aatoolbox/pipeline.py [hunk 5]
```diff
@@ -101,3 +109,91 @@ def _load_codab(self, layer_name: str, clobber: bool):
...
+        return GeoBoundingBox.from_shape(gdf)
+
+    def load_geoboundingbox_coordinates(self, coordinates: Coordinates):
```
[the remark this answers is not in the document]
Comment (Tinkaa) [reply src/aatoolbox/pipeline.py:141]: Hehe yes smart. Was because of previous structure, but doesn't make any sense now anymore.

File: tests/pipeline/test_iri_pipeline.py (hunk 3 above)
Comment (Tinkaa) [reply tests/pipeline/test_iri_pipeline.py:1]: Hmm yes I was first thinking about this as well. But then felt like we should keep pipeline and the main functionalities separate. And therefore I would argue to also keep the tests of them separate, else it gets pretty diluted. Why would you say we should have them all in the pipeline file?

File: src/aatoolbox/pipeline.py [hunk 6]
```diff
@@ -101,3 +109,91 @@ def _load_codab(self, layer_name: str, clobber: bool):
...
+        ... load_geoboundingbox_coordinates(coordinates))
+        """
+        return GeoBoundingBox(
+            north=coordinates["north"],
+            south=coordinates["south"],
+            east=coordinates["east"],
+            west=coordinates["west"],
+        )
+
+    def load_iri_forecast_probability(
```
[the remark this answers is not in the document]
Comment (Tinkaa) [reply src/aatoolbox/pipeline.py:164]: Yeah I am struggling with this. The thing is that the `dominant` one is also probabilistic.. I would optimally call the two `load_iri_seasonal_tercile_forecast` and `load_iri_seasonal_dominant_tercile_forecast` . Would that be more clear to you and acceptabel length?

[the remark this answers is not in the document]
Comment (turnerm) [reply src/aatoolbox/pipeline.py:155]: Yes I like that better! But then what's the difference between `load_iri_seasonal_tercile_forecast` and `load_iri_seasonal_dominant_tercile_forecast` -- is it that the first one loads *all* terciles? Maybe `all` or `full` should be in the name then

File: tests/pipeline/test_iri_pipeline.py (hunk 3 above)
Comment (turnerm) [reply tests/pipeline/test_iri_pipeline.py:1]: Oh well I was thinking if you go through `pipeline` then you can test everything through there and don't have to write duplicates. But since the future of pipeline is uncertain then you can leave it for now

File: src/aatoolbox/pipeline.py [hunk 7]
```diff
@@ -101,3 +109,91 @@ def _load_codab(self, layer_name: str, clobber: bool):
...
+        self, geo_bounding_box, clobber: bool = False
+    ):
+        """
+        Load the IRI seasonal tercile forecast.
+
+        This data contains a probability per tercile.
+        """
+        iri_forecast_probability = IriForecastProb(
```
[the remark this answers is not in the document]
Comment (turnerm) [reply src/aatoolbox/pipeline.py:163]: Hm -- what if you type it as `_IriForecast` so it's happy with either type? But anyway don't worry too much

File: src/aatoolbox/pipeline.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Tinkaa) [reply src/aatoolbox/pipeline.py:155]: Yeah exactly that, the first one loads the probability per tercile and the other one only one probability, which indicates that of the dominant tercile. will rename to `load_iri_seasonal_all_terciles_forecast` and `load_iri_seasonal_dominant_tercile_forecast`

File: src/aatoolbox/pipeline.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Tinkaa) [reply src/aatoolbox/pipeline.py:165]: :/ not working but gonna leave it
